pennylane/templates/embeddings/iqp.py

```diff
@@ -220,22 +181,46 @@ def circuit(features=None, pattern=None):
 
     """
 
-    wires = Wires(wires)
-    pattern = _preprocess(features, wires, pattern)
+    num_params = 1
+    num_wires = AnyWires
+    par_domain = "A"
+
+    def __init__(self, features, wires, n_repeats=1, pattern=None, do_queue=True):
+
+        shape = qml.math.shape(features)
+
+        if len(shape) != 1:
+            raise ValueError(f"Features must be a one-dimensional tensor; got shape {shape}.")
+
+        n_features = shape[0]
+        if n_features != len(wires):
+            raise ValueError(f"Features must be of length {len(wires)}; got length {n_features}.")
+
+        if pattern is None:
+            # default is an all-to-all pattern
+            pattern = [wire_pair for wire_pair in combinations(wires, 2)]
```

Comment:
Another subtle change: I am not parsing wires to `Wires` here, since this will be done when the gate is called with the pair of wires anyways. Should remove a bit of overhead.